internal/api/utils.go

```diff
@@ -17,6 +18,20 @@ func DeleteNilsFromPtrSlice[S ~[]*E, E any](s S) S {
 	return slices.DeleteFunc(s, func(e *E) bool { return e == nil })
 }
 
+// NonNil returns a iterator over index-value pairs in a slice
+// of pointers in the usual order, but skipping over nils.
+func NonNil[E any](a []*E) iter.Seq2[int, *E] {
+	return func(yield func(int, *E) bool) {
+		for i := range a {
+			if a[i] != nil {
+				if !yield(i, a[i]) {
+					return
+				}
+			}
+		}
+	}
+}
```

Comment:
This is great!  Would you mind renaming it to something like `NonNilSlice` because I'd like to write a similar function for maps.

Reply:
Updated to `NonNilSliceValues`.